apple/RNGHVector.h

```diff
@@ -0,0 +1,29 @@
+//
+//  RNGHVector.h
+//  Pods
+//
+//  Created by Michał Bert on 05/08/2024.
+//
+
+#import "RNGestureHandlerDirection.h"
+
+#ifndef RNGHVector_h
+#define RNGHVector_h
+
+@interface Vector : NSObject
+
+@property (atomic, assign) double x;
+@property (atomic, assign) double y;
+@property (atomic, assign) double unitX;
+@property (atomic, assign) double unitY;
+@property (atomic, assign) double magnitude;
+
++ (Vector *)fromDirection:(RNGestureHandlerDirection)direction;
++ (Vector *)fromVelocityX:(double)vx withVelocityY:(double)vy;
+- initWithX:(double)x withY:(double)y;
```

Comment:
```suggestion
- (nonnull instancetype)initWithX:(double)x withY:(double)y;
```

It's likely that the signature would also need to be updated in the implementation.

Reply:
Done in [3fcac48](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/3fcac48d3af08b460004341f0c62767de48a71ee)

Reply:
Done in [3fcac48](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/3fcac48d3af08b460004341f0c62767de48a71ee). 

IDE also pointed out that I have to update other signatures and specify whether the pointers can be null or not.